integration/class-classic-editor.php

```diff
@@ -18,73 +20,188 @@ class Classic_Editor {
-		if ( ! isset( $_POST['activitypub_quote_policy_nonce'] ) ) {
+		// Only process for post types that support ActivityPub.
+		if ( ! \post_type_supports( \get_post_type( $post_id ), 'activitypub' ) ) {
 			return;
 		}
 
-		// Verify nonce.
-		if ( ! \wp_verify_nonce( \sanitize_text_field( \wp_unslash( $_POST['activitypub_quote_policy_nonce'] ) ), 'activitypub_quote_policy' ) ) {
+		// Check user permissions.
+		if ( ! \current_user_can( 'edit_post', $post_id ) ) {
 			return;
 		}
 
-		// Only process for post types that support ActivityPub.
-		if ( ! \post_type_supports( \get_post_type( $post_id ), 'activitypub' ) ) {
+		// Verify nonce is present and valid.
+		if ( ! isset( $_POST['activitypub_meta_box_nonce'] ) ) {
 			return;
 		}
 
-		// Check user permissions.
-		if ( ! \current_user_can( 'edit_post', $post_id ) ) {
+		if ( ! \wp_verify_nonce( \sanitize_text_field( \wp_unslash( $_POST['activitypub_meta_box_nonce'] ) ), 'activitypub_meta_box' ) ) {
 			return;
 		}
 
-		// Save quote interaction policy based on checkbox.
-		if ( '1' === \sanitize_text_field( \wp_unslash( $_POST['activitypub_allow_quotes'] ?? '0' ) ) ) {
-			// Checked: allow anyone to quote (default value).
-			\update_post_meta( $post_id, 'activitypub_interaction_policy_quote', ACTIVITYPUB_INTERACTION_POLICY_ANYONE );
-		} else {
-			// Unchecked: only allow "just me" to quote.
-			\update_post_meta( $post_id, 'activitypub_interaction_policy_quote', ACTIVITYPUB_INTERACTION_POLICY_ME );
+		// Save content warning.
+		if ( isset( $_POST['activitypub_content_warning'] ) ) {
+			$content_warning = \sanitize_text_field( \wp_unslash( $_POST['activitypub_content_warning'] ) );
+			if ( ! empty( $content_warning ) ) {
+				\update_post_meta( $post_id, 'activitypub_content_warning', $content_warning );
+			} else {
+				\delete_post_meta( $post_id, 'activitypub_content_warning' );
+			}
+		}
+
+		// Save max image attachments.
+		if ( isset( $_POST['activitypub_max_image_attachments'] ) ) {
+			$max_images = \absint( $_POST['activitypub_max_image_attachments'] );
+			if ( $max_images >= 0 && $max_images <= 10 ) {
+				\update_post_meta( $post_id, 'activitypub_max_image_attachments', $max_images );
+			}
+		}
+
+		// Save content visibility.
+		if ( isset( $_POST['activitypub_content_visibility'] ) ) {
+			$visibility = \sanitize_text_field( \wp_unslash( $_POST['activitypub_content_visibility'] ) );
+			if ( \in_array( $visibility, array( 'public', 'quiet_public', 'local' ), true ) ) {
+				\update_post_meta( $post_id, 'activitypub_content_visibility', $visibility );
+			}
+		}
+
+		// Save quote interaction policy.
+		if ( isset( $_POST['activitypub_interaction_policy_quote'] ) ) {
+			$quote_policy = \sanitize_text_field( \wp_unslash( $_POST['activitypub_interaction_policy_quote'] ) );
+			if ( \in_array( $quote_policy, array( 'anyone', 'followers', 'me' ), true ) ) {
+				\update_post_meta( $post_id, 'activitypub_interaction_policy_quote', $quote_policy );
+			}
```

Comment:
This could probably be simplified to a meta key check in `$_POST` and then running `update_post_meta()` with it. Sanitization and handling of empty values should be covered by the register_meta calls and filters.